### DISCUSSION TITLE
new sensor for filter

### DISCUSSION THREAD

**Discussion by pet-valent** [davidm-glitch/home-assistant-eheim-digital#14] [Ideas]
It would be nice to have a sensor that return value in days that remains to the filter service as a part of the integration

filter_XXX_next service minus "now" rounded to days

by this it would be possible to observe in the graph or any other type of display the speed of the filter pollution in a time

or is the remaining days info already available in the integration and then converted to the next service entity?

**Comment by davidm-glitch** [comment]
The remaining days are available from the api, we calculate the date with this. You could try creating your own template sensor to make your calculations.

I will check later how to do this.

**Comment by davidm-glitch** [comment]
This is what your template sensor should look like to get the remaining days:

`{{ ((as_timestamp(states('sensor.eheim_filter_700_next_service')) | float - as_timestamp(now()) | float) / 86400) | round}}`

**Reply by pet-valent** [reply]
Thank you, this works. Idea was to not calculate value up and down if it is available by API, since result should be still tha same
